test/unit/mem/view/src/ViewSubViewTest.cpp

```diff
@@ -266,4 +313,179 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(
+    Stream stream (devAcc);
+
+    alpaka::Vec<Dim, Size> elementsPerThread(alpaka::Vec<Dim, Size>::all(static_cast<Size>(1)));
+    alpaka::Vec<Dim, Size> threadsPerBlock(alpaka::Vec<Dim, Size>::all(static_cast<Size>(1)));
+    alpaka::Vec<Dim, Size> const blocksPerGrid(alpaka::Vec<Dim, Size>::all(static_cast<Size>(1)));
+
+    WorkDiv const workdiv(alpaka::workdiv::WorkDivMembers<Dim, Size>(
+        blocksPerGrid,
+        threadsPerBlock,
+        elementsPerThread));    
+
+
+    auto const extentBuf(alpaka::Vec<Dim, Size>::all(nElementsPerDim));
+    auto const extentView(alpaka::Vec<Dim, Size>::all(nElementsPerDimView));
+    auto const offsetView(alpaka::Vec<Dim, Size>::all(offsetInAllDims));
+    auto buf(alpaka::mem::buf::alloc<Elem, Size>(devAcc, extentBuf));
+    auto buf2(alpaka::mem::buf::alloc<Elem, Size>(devAcc, extentView));    
+    View view(buf, extentView, offsetView);
+
+    // Init buf with increasing values
+    std::vector<Elem> v(extentBuf.prod(), static_cast<Elem>(0));
+    std::iota(v.begin(), v.end(), static_cast<Elem>(0));
+    ViewPlainPtr plainBuf(v.data(), devHost, extentBuf);
+    alpaka::mem::view::copy(stream, buf, plainBuf, extentBuf);    
+
+    CompareBufferKernel compareBufferKernel;
+
+    switch(Dim::value){
+    case 1:
+        {
+        std::vector<Elem> v2{1,2};
+        ViewPlainPtr plainBuf2(v2.data(), devHost, extentView);
+        alpaka::mem::view::copy(stream, buf2, plainBuf2, extentView);            
+        auto const compare (alpaka::exec::create<TAcc> (
+            workdiv,
+            compareBufferKernel,
+            alpaka::test::mem::view::begin(buf2),
+            alpaka::test::mem::view::end(buf2),
+            alpaka::test::mem::view::begin(view))); 
+        break;
+        }
+    case 2:
+        {
+        std::vector<Elem> v2{5, 6, 9, 10};
+        ViewPlainPtr plainBuf2(v2.data(), devHost, extentView);
+        alpaka::mem::view::copy(stream, buf2, plainBuf2, extentView);            
+        auto const compare (alpaka::exec::create<TAcc> (
+            workdiv,
+            compareBufferKernel,
+            alpaka::test::mem::view::begin(buf2),
+            alpaka::test::mem::view::end(buf2),
+            alpaka::test::mem::view::begin(view))); 
+        break;
+        }
+    case 3:
+        {
+        std::vector<Elem> v2{21, 22, 25, 26, 37, 38, 41, 42};
+        ViewPlainPtr plainBuf2(v2.data(), devHost, extentView);
+        alpaka::mem::view::copy(stream, buf2, plainBuf2, extentView);            
+        auto const compare (alpaka::exec::create<TAcc> (
+            workdiv,
+            compareBufferKernel,
+            alpaka::test::mem::view::begin(buf2),
+            alpaka::test::mem::view::end(buf2),
+            alpaka::test::mem::view::begin(view))); 
+        break;
+        }
+    case 4:
+        {
+        std::vector<Elem> v2{75, 76, 78, 79, 91, 92, 95, 96, 139, 140, 143, 144, 155, 156, 159, 160};
```

Comment:
For now it would be good if you could comment out the 4 dimensional case here and use `if(Dim::value != 4u)` in the other test case. This would allow us to merge this independently of a 4D copy support.